### ISSUE TITLE
Need to specify DASHBOARD_TOKEN env var

### ISSUE THREAD

**Issue by suryaveer** [artillery/pull-request-viewer#2]
Hey guys, I found this project particularly helpful, and was trying to set it up locally on my system. I set the .env file with all the 4 required key-value pairs successfully, but the app never loads for me.

This is what I see at startup
```
user1:~/dev/personal/pull-request-viewer $ npm run dev

> pull-request-viewer@0.0.1 dev /Users/user1/dev/personal/pull-request-viewer
> nodemon ./prviewer.coffee settings.json

11 Jan 22:07:20 - [nodemon] v1.2.1
11 Jan 22:07:20 - [nodemon] to restart at any time, enter `rs`
11 Jan 22:07:20 - [nodemon] watching: *.*
11 Jan 22:07:20 - [nodemon] starting `coffee ./prviewer.coffee settings.json`
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CALLBACK_URL="https://github.com/auth/github_oauth/callback"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CLIENT_ID="NNNNNNNNNNNNN"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CLIENT_SECRET="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_REPOS="foo/bar1,foo/bar2,foo1/bar3"
Forced reviewers: {}
Username aliases: {}
Pull Request Viewer listening on http://localhost:8000
Need to specify DASHBOARD_TOKEN env var
11 Jan 22:07:30 - [nodemon] app crashed - waiting for file changes before starting...
```

Any suggestions on how to proceed forward?

**Comment by statico** [comment]
Sorry about that! Should be fixed in efc00d8412fc666bd0a403ef4ea183c73eafec0e

**Comment by suryaveer** [comment]
Thanks for the quick fix, but still I can't get the auth to work with using localhost. When I try the heroku method, I run into other npm version issues

```
~/dev/personal/pull-request-viewer $ git push heroku
Counting objects: 8, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (6/6), done.
Writing objects: 100% (6/6), 591 bytes | 0 bytes/s, done.
Total 6 (delta 4), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Node.js app detected
remote: 
remote:        Node engine:         0.8.x
remote:        Npm engine:          1.1.x
remote:        Start mechanism:     Procfile
remote:        node_modules source: package.json
remote:        node_modules cached: true
remote: 
remote:        NPM_CONFIG_PRODUCTION=true
remote:        NODE_MODULES_CACHE=true
remote: 
remote: -----> Installing binaries
remote:        Resolving node version 0.8.x via semver.io...
remote:        Downloading and installing node 0.8.28...
remote:        Resolving npm version 1.1.x via semver.io...
remote:        Downloading and installing npm 1.1.71 (replacing version 1.2.30)...
remote: 
remote: -----> Building dependencies
remote:        Npm version changed (1.2.8000 => 1.1.71); invalidating cache
remote:        Installing node modules
remote:        npm WARN package.json methods@0.0.1 No README.md file found!
remote:        npm WARN engine void-elements@1.0.0: wanted: {"node":">=0.10.0"} (current: {"node":"v0.8.28","npm":"1.1.71"})
remote:        npm ERR! Error: No compatible version found: acorn-globals@'^1.0.1'
remote:        npm ERR! Valid install targets:
remote:        npm ERR! ["1.0.0","1.0.1"]
remote:        npm ERR!     at installTargetsError (/tmp/build_135befd8cc671633237571061eee935e/.heroku/node/lib/node_modules/npm/lib/cache.js:564:10)
remote:        npm ERR!     at /tmp/build_135befd8cc671633237571061eee935e/.heroku/node/lib/node_modules/npm/lib/cache.js:479:10
remote:        npm ERR!     at saved (/tmp/build_135befd8cc671633237571061eee935e/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/get.js:138:7)
remote:        npm ERR!     at /tmp/build_135befd8cc671633237571061eee935e/.heroku/node/lib/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:218:7
remote:        npm ERR!     at Object.oncomplete (fs.js:302:15)
remote:        npm ERR! If you need help, you may report this log at:
remote:        npm ERR!     <http://github.com/isaacs/npm/issues>
remote:        npm ERR! or email it to:
remote:        npm ERR!     <npm-@googlegroups.com>
remote:        
remote:        npm ERR! System Linux 3.8.11-ec2
remote:        npm ERR! command "node" "/tmp/build_135befd8cc671633237571061eee935e/.heroku/node/bin/npm" "install" "--quiet" "--userconfig" "/tmp/build_135befd8cc671633237571061eee935e/.npmrc"
remote:        npm ERR! cwd /tmp/build_135befd8cc671633237571061eee935e
remote:        npm ERR! node -v v0.8.28
remote:        npm ERR! npm -v 1.1.71
remote:        npm ERR! 
remote:        npm ERR! Additional logging details can be found in:
remote:        npm ERR!     /tmp/build_135befd8cc671633237571061eee935e/npm-debug.log
remote:        npm ERR! not ok code 0
remote: 
remote: -----> Build failed
remote: 
remote:        WARNING: This version of npm (1.1.71) has several known issues - consider upgrading to the latest release (2.1.18)
remote:        https://devcenter.heroku.com/articles/nodejs-support#specifying-an-npm-version
remote:        
```
Upgrading npm to 1.2.x, or even to 2.1.x doesn't work, because I end up getting a host of other warnings!

Do you have any suggestions on getting it to build locally from scratch? I'm new to npm and node.

**Comment by statico** [comment]
Please `git pull` and give it another try! (Specifically, I would compare the contents of `.env` to my example below, `rm -rf node_modules`, then `npm install`, then `npm run dev`.)

I bumped the required version up to Node 0.10.x and left the npm version unspecified, which might help you. I did a fresh `npm install` with Node 0.10.32 as well as a new push to Heroku, and both succeeded.

Here's my local environment:

```
ian ▸ node --version
v0.10.32
ian ▸ npm --version
2.1.18
```

And here's most of my local `.env`:

```
GITHUB_REPOS=ahgl/ahgl-site
GITHUB_FORCE_REVIEWERS=ahgl/ahgl-site:statico
GITHUB_CLIENT_ID=xxxxxxxxxxxxxxxxx
GITHUB_CLIENT_SECRET=xxxxxxxxxxxxxxxxx
GITHUB_CALLBACK_URL=http://localhost:8000/auth/github/callback
GITHUB_USERNAME_ALIASES=ian:statico
```

**Comment by suryaveer** [comment]
Thanks, I'll give it a try. Did you set the Homepage URL in github application as "localhost:8000" ..?

**Comment by suryaveer** [comment]
Also, it was failing on heroku, not local machine. That stack trace was from remote
```
remote:        WARNING: This version of npm (1.1.71) has several known issues - consider upgrading to the latest release (2.1.18)
remote:        https://devcenter.heroku.com/articles/nodejs-support#specifying-an-npm-version
```

But I'll give this another try sometime later today. Thanks for being so responsive!

**Comment by suryaveer** [comment]
Setting the .env as your still gives me issues when running locally (haven't tried on heroku yet):
![screen shot 2015-01-13 at 11 58 58 am](https://cloud.githubusercontent.com/assets/3867082/5727986/c5e75ace-9b1b-11e4-82be-5beb318657ac.png)

**Comment by statico** [comment]
Ah, the authorization callback should point to either localhost or Heroku.

Here's our Heroku app config:
![image](https://cloud.githubusercontent.com/assets/137158/5744577/cc7f588a-9bd6-11e4-8b96-47edcee3ee1b.png)

And here's the config for localhost:
![image](https://cloud.githubusercontent.com/assets/137158/5744665/97628068-9bd7-11e4-9b47-ff169cf6f5f9.png)

**Comment by statico** [comment]
(I should probably add this to the README...)

**Comment by suryaveer** [comment]
Let me try this real quick .. I had tried the localhost config settings .. but that didn't work earlier...

**Comment by suryaveer** [comment]
So, the localhost setting doesn't work for me:

```
~/dev/personal/pull-request-viewer $ npm run dev
> pull-request-viewer@0.0.1 dev /Users/user1/dev/personal/pull-request-viewer
> nodemon ./prviewer.coffee settings.json

14 Jan 10:28:34 - [nodemon] v1.2.1
14 Jan 10:28:34 - [nodemon] to restart at any time, enter `rs`
14 Jan 10:28:34 - [nodemon] watching: *.*
14 Jan 10:28:34 - [nodemon] starting `coffee ./prviewer.coffee settings.json`
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CALLBACK_URL="http://localhost:8000/auth/github/callback"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CLIENT_ID="xxxxxxxxxx"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_CLIENT_SECRET="xxxxxxxx"
Read from /Users/user1/dev/personal/pull-request-viewer/.env: GITHUB_REPOS="suryaveer/sql2es,suryaveer/elseql"
Forced reviewers: {}
Username aliases: {}
Pull Request Viewer listening on http://localhost:8000
GET / 302 6ms - 80
GET /auth/github 302 3ms - 0
```
![screen shot 2015-01-14 at 10 31 33 am](https://cloud.githubusercontent.com/assets/3867082/5744798/a033f68a-9bd8-11e4-8ddb-cce1a23104c3.png)

And when I hit localhost:8000, I get the same 404 as above.

**Comment by suryaveer** [comment]
I could however get the heroku app to work .. One last question - if I want this to work for a private github deployment .. not github.com .. is there an easy way to do that?

**Comment by statico** [comment]
Ah. I think you have quotes in your `.env` file. Try removing the double quotes from the settings.

**Comment by suryaveer** [comment]
For a private github deployment, it still routes to "https://github.com/login/oauth/authorize?<>", which I guess I'll need to change to point to my server's auth end point. I'm just curious if you guys have already done it for your private deployment..

**Comment by statico** [comment]
We don't have an enterprise GitHub account, so I haven't used the viewer in that environment. However, we're using the [github npm module](https://www.npmjs.com/package/github), which in its README seems to support setting the host. Feel free to send me a pull request where the GitHub client is initialized from a new environment variable.

**Comment by suryaveer** [comment]
Sure, will do! Thanks for the help!
